pipeline/directed.go

```diff
@@ -11,18 +11,15 @@ import (
 	"gonum.org/v1/gonum/graph/topo"
 )
 
+var _ Pipeline = (*DirectedPipeline)(nil)
```

Comment:
Do we have any protection against plugin nesting cycles?

`plugin_a -> plugin_b -> plugin_a`

Reply:
No, we currently do not. Do you think it's worth the complexity? Since plugins are maintained by us, it felt like something that we should just avoid.

Reply:
We do expect that users will modify/write plugins, so I think the question of how important this is comes down to how bad of a user experience this will be when it happens.

If it's significantly easier, I'd be ok with a max depth of 5 or 10, or whatever seems reasonable. But I tend to think we should prevent the known runaway scenario.

Reply:
@djaglowski Turned out to be way easier than expected. Pushed up the change in the latest commit